pkg/clustermesh/endpointslicesync/endpointslice_test.go

```diff
@@ -181,12 +169,13 @@ func Test_meshEndpointSlice_Reconcile(t *testing.T) {
 		svcStore.CacheStore().Add(svc1)
 		serviceInformer.refreshAllCluster(svc1)
 
-		queue := getControllerQueue(controller)
-		require.NoError(t, waitEmptyQueue(queue))
+		var epList *discovery.EndpointSliceList
+		require.EventuallyWithT(t, func(c *assert.CollectT) {
+			epList, err = getEndpointSlice(&fakeClient, svcName)
+			assert.NoError(c, err)
+			assert.Len(t, epList.Items, 1)
```

Comment:
```suggestion
			assert.Len(c, epList.Items, 1)
```

Otherwise the test will incorrectly fail if the assertion fails once.